src/pages/vue/your-first-app/2-taking-photos.md

```diff
@@ -0,0 +1,170 @@
+---
+previousText: 'Your First App'
+previousUrl: '/docs/vue/your-first-app'
+nextText: 'Saving Photos on Filesystem'
+nextUrl: '/docs/vue/your-first-app/3-saving-photos'
+---
+
+# Taking Photos with the Camera
+
+Now for the fun part - adding the ability to take photos with the device’s camera using the Capacitor [Camera API](https://capacitor.ionicframework.com/docs/apis/camera). We’ll begin with building it for the web, then make some small tweaks to make it work on mobile (iOS and Android).
+
+To do so, we will create a standalone composition function paired with Vue's Composition API to manage the photos for the gallery.
+
+> If you are not familiar with Vue's Composition API, [Why Composition API?](https://v3.vuejs.org/guide/composition-api-introduction.html#why-composition-api) from the official Vue docs is a good resource to start with.
+
+Create a new file at `src/composables/usePhotoGallery.ts` and open it up.
+
+We will start by importing the various utilities we will use from Vue core, Ionic Vue, and Capacitor:
+
+```typescript
+import { ref, onMounted, watch } from 'vue';
+import { Plugins, CameraResultType, CameraSource, CameraPhoto, 
+Capacitor, FilesystemDirectory } from "@capacitor/core";
+import { isPlatform } from '@ionic/vue';
```

Comment:
We might want to hold off on importing this right now as it appears we do not use it until later. If users try to test the app after this point, the compiler is going to give a warning on the screen that `isPlatform` is un-used.